src/lightning/data/cache/serializers.py

```diff
@@ -0,0 +1,181 @@
+# Copyright The Lightning AI team.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from io import BytesIO
+
+import numpy as np
+import torch
+from lightning_utilities.core.imports import RequirementCache
+
+_PIL_AVAILABLE = RequirementCache("PIL")
+
+if _PIL_AVAILABLE:
+    from PIL import Image
+    from PIL.JpegImagePlugin import JpegImageFile
+else:
+    Image = None
+    JpegImageFile = None
+
```

Comment:
In general, can we please try to use pickle (or anything else, but pickle is python-native) for everything we can? Otherwise this really becomes too datatype specific and we'll always get issues about unimplemented types.

Reply:
pickle isn't a good format. Using raw format makes the dataset too bit. In order to make things fast, you need to use optimised format like JPEG.

Reply:
There's probably  a reason why pytorch opted for pickling + zip in their tensor serialization. That should reduce it as much as possible, no?

Reply:
No, I have checked it out. It is bigger and slower.